### PR TITLE
Port all plugins to use the window-construct property

### PR DESCRIPTION
**Pull request by mbkma**
pluma-file-browser-plugin.c: rename "data" to "priv".

Do not merge before #615

### CODE REVIEW

File: plugins/docinfo/pluma-docinfo-plugin.c [hunk 1]
```diff
@@ -505,8 +505,8 @@ pluma_docinfo_plugin_set_property (GObject      *object,
 
 	switch (prop_id)
 	{
-		case PROP_OBJECT:
-			plugin->priv->window = GTK_WIDGET (g_value_dup_object (value));
+		case PROP_WINDOW:
+			plugin->priv->window = PLUMA_WINDOW (g_value_dup_object (value));
```
Comment (rbuj) [inline plugins/docinfo/pluma-docinfo-plugin.c:509]:
```
pluma-docinfo-plugin.c:509:25: warning: assignment to ‘GtkWidget *’ {aka ‘struct _GtkWidget *’} from incompatible pointer type ‘PlumaWindow *’ {aka ‘struct _PlumaWindow *’} [-Wincompatible-pointer-types]
```

File: plugins/filebrowser/pluma-file-browser-plugin.c [hunk 2]
```diff
@@ -154,8 +155,8 @@ pluma_file_browser_plugin_set_property (GObject      *object,
 
 	switch (prop_id)
 	{
-		case PROP_OBJECT:
-			plugin->priv->window = GTK_WIDGET (g_value_dup_object (value));
+		case PROP_WINDOW:
+			plugin->priv->window = PLUMA_WINDOW (g_value_dup_object (value));
```
Comment (rbuj) [inline plugins/filebrowser/pluma-file-browser-plugin.c:159]:
```
pluma-file-browser-plugin.c:159:25: warning: assignment to ‘GtkWidget *’ {aka ‘struct _GtkWidget *’} from incompatible pointer type ‘PlumaWindow *’ {aka ‘struct _PlumaWindow *’} [-Wincompatible-pointer-types]
```

File: plugins/sort/pluma-sort-plugin.c [hunk 3]
```diff
@@ -461,8 +461,8 @@ pluma_sort_plugin_set_property (GObject      *object,
 
 	switch (prop_id)
 	{
-		case PROP_OBJECT:
-			plugin->priv->window = GTK_WIDGET (g_value_dup_object (value));
+		case PROP_WINDOW:
+			plugin->priv->window = PLUMA_WINDOW (g_value_dup_object (value));
```
Comment (rbuj) [inline plugins/sort/pluma-sort-plugin.c:465]:
```
pluma-sort-plugin.c:465:25: warning: assignment to ‘GtkWidget *’ {aka ‘struct _GtkWidget *’} from incompatible pointer type ‘PlumaWindow *’ {aka ‘struct _PlumaWindow *’} [-Wincompatible-pointer-types]
```

**Review by rbuj — COMMENTED** [review]
There are 3 new incompatible-pointer-types warnings:

File: plugins/docinfo/pluma-docinfo-plugin.c (hunk 1 above)
Comment (rbuj) [reply plugins/docinfo/pluma-docinfo-plugin.c:509]:
```diff
diff --git a/plugins/docinfo/pluma-docinfo-plugin.c b/plugins/docinfo/pluma-docinfo-plugin.c
index 4bb2633..df98b34 100644
--- a/plugins/docinfo/pluma-docinfo-plugin.c
+++ b/plugins/docinfo/pluma-docinfo-plugin.c
@@ -60,7 +60,7 @@ typedef struct
 
 struct _PlumaDocInfoPluginPrivate
 {
-       GtkWidget *window;
+       PlumaWindow *window;
 
        GtkActionGroup *ui_action_group;
        guint ui_id;
```

File: plugins/filebrowser/pluma-file-browser-plugin.c (hunk 2 above)
Comment (rbuj) [reply plugins/filebrowser/pluma-file-browser-plugin.c:159]:
```diff
diff --git a/plugins/filebrowser/pluma-file-browser-plugin.c b/plugins/filebrowser/pluma-file-browser-plugin.c
index ea35287..5212bdc 100644
--- a/plugins/filebrowser/pluma-file-browser-plugin.c
+++ b/plugins/filebrowser/pluma-file-browser-plugin.c
@@ -53,7 +53,7 @@
 
 struct _PlumaFileBrowserPluginPrivate
 {
-       GtkWidget               *window;
+       PlumaWindow             *window;
 
        PlumaFileBrowserWidget * tree_widget;
        gulong                   merge_id;
```

File: plugins/sort/pluma-sort-plugin.c (hunk 3 above)
Comment (rbuj) [reply plugins/sort/pluma-sort-plugin.c:465]:
```diff
diff --git a/plugins/sort/pluma-sort-plugin.c b/plugins/sort/pluma-sort-plugin.c
index 06b379f..2713cab 100644
--- a/plugins/sort/pluma-sort-plugin.c
+++ b/plugins/sort/pluma-sort-plugin.c
@@ -61,7 +61,7 @@ typedef struct
 
 struct _PlumaSortPluginPrivate
 {
-       GtkWidget *window;
+       PlumaWindow *window;
 
        GtkActionGroup *ui_action_group;
        guint ui_id;
```

File: plugins/spell/pluma-spell-plugin.c [hunk 4]
```diff
@@ -1384,7 +1384,7 @@ pluma_spell_plugin_set_property (GObject      *object,
 
 	switch (prop_id)
 	{
-		case PROP_OBJECT:
+		case PROP_WINDOW:
 			plugin->priv->window = GTK_WIDGET (g_value_dup_object (value));
```
Comment (rbuj) [inline plugins/spell/pluma-spell-plugin.c]: plugin->priv->window = PLUMA_WINDOW (g_value_dup_object (value));

**Review by rbuj — COMMENTED** [review]
Review the setter method and the window property.
```diff
diff --git a/plugins/spell/pluma-spell-plugin.c b/plugins/spell/pluma-spell-plugin.c
index 0b5dc5a..8d916dd 100644
--- a/plugins/spell/pluma-spell-plugin.c
+++ b/plugins/spell/pluma-spell-plugin.c
@@ -63,7 +63,7 @@ enum {
 
 struct _PlumaSpellPluginPrivate
 {
-       GtkWidget *window;
+       PlumaWindow *window;
 
        GtkActionGroup *action_group;
        guint           ui_id;
```

**Review by rbuj — APPROVED** [review]
LGTM

**Comment by rbuj** [comment]
```
$  git rebase master
```

**Review by raveit65 — APPROVED** [review]
Builds and runs fine in a quick test.

**Comment by raveit65** [comment]
Ok, anything left to merge before a new release?

**Comment by rbuj** [comment]
I think that's it.

**Pull request merged**